### ISSUE TITLE
Can't Importing Components from Internal Packages with Federation on Dev mode

### ISSUE THREAD

**Issue by harrisKAsher** [module-federation/vite#187]
A remote app when I import a component from a package (ComponentA) that internally imports another component (ComponentB) from a dependent package, it fails to load the entire federated remote app. If I already import ComponentB in the remote app then everything works. This issue doesn't occur when running I go to the remote apps endpoint and view it outside the federated instance. It also doesn't happen if I do a build and preview it in the Federated Instance. I have tried a bunch of different adjustments to make ComponentA and/or ComponentB shared with the vite config but none that changes anything.

I was trying to reproduce this and I cannot do it without one thing: getting from a private npm repository. My organization has an internal package of components to make things uniform across the various web applications we make. Its kind of similar to [HeadlessUI](https://headlessui.com/). Anyway, that is what is breaking the app, and some of them are pretty simple. Literally some of them are just div's with styles on them. But it seems that if an internal package relies on another internal package that is when it breaks.

My only theory is it is breaking because of how the federation in dev mode can't get the internal package for some reason? Not really sure how to verify that though as I am not that familiar with internal packages. I tried with other packages that are similar such as HeadlessUI but I could not get it to break in the same way.

**Comment by harrisKAsher** [comment]
I am closing this, I have done more testing and I have figured out it has something to do with my organizations internal package and I have no way to reproduce it. I have also found a way to work around it.

**Comment by harrisKAsher** [comment]
Just in case anyone ever find this off google and wonders about the work around I did was.

Basically I just made a component and stuck all of the dependency components (ComponentB) inside it. Then in my app I added this `{false && <DependencyComponent />}` This makes the app think it could be loaded and so it bundles the necessary components in the dev mode while not showing or using those components.
